pkg/apiserver/registry/intelligence/networkpolicyrecommendation/rest.go

```diff
@@ -60,6 +69,15 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 	}
 	intelliNPR := new(intelligence.NetworkPolicyRecommendation)
 	r.copyNetworkPolicyRecommendation(intelliNPR, npReco)
+	// Try to retrieve result from ClickHouse in case NPR is completed
+	if npReco.Status.State == crdv1alpha1.NPRecommendationStateCompleted {
+		result, err := r.getRecommendationResult(npReco.Status.SparkApplication)
+		if err != nil {
+			klog.Warningf("Failed to get the result for completed NetworkPolicy Recommedation with id %s, error: %v", npReco.Status.SparkApplication, err)
```

Comment:
do you think it might be worth populating ErrorMsg in the CRD status if we fail to fetch the results from CH?

Reply:
I updated to populate the ErrorMsg. My only concern is usually we keep consistency between `status.State` and `status.ErrorMsg`, i.e. `status.ErrorMsg` will be empty if the `status.State` is not failed. But in the case of failing to fetch the results from CH, we need to choose between

- break the consistency, i.e. we may have a State of Completed with some ErrorMsg
- overwrite the State value which indicates the status of the SparkApplication 

Currently I choose the first option. Please tell me if you have other idea for this.